### PR TITLE
Refactor wave solver files: Split field namespace for each physics

### PR DESCRIPTION
**Pull request by acitrain**
This PR is the first part of a global refactoring for wave solvers (it will be followed with, at least, three other) 

This part contains the splitting of WaveSolverBaseFields namespace into a specific namespace for the different physics (acoustic, elastic,acousto-elastic) and some clean up 

New baseline: GEOS-DEV/integratedTests/pull/93

### CODE REVIEW

**Review by tbeltzun — APPROVED** [review]

**Review by mndiaye24 — APPROVED** [review]
Nice work, field definitions become more clear !

File: src/coreComponents/physicsSolvers/wavePropagation/AcousticFirstOrderWaveEquationSEM.hpp [hunk 1]
```diff
@@ -75,12 +69,12 @@ class AcousticFirstOrderWaveEquationSEM : public WaveSolverBase
...
   /**@}*/
 
-  /**
-   * @brief Multiply the precomputed term by the Ricker and add to the right-hand side
-   * @param cycleNumber the cycle number/step number of evaluation of the source
-   * @param rhs the right hand side vector to be computed
-   */
-  virtual void addSourceToRightHandSide( integer const & cycleNumber, arrayView1d< real32 > const rhs );
+  // /**
+  //  * @brief Multiply the precomputed term by the Ricker and add to the right-hand side
```
Comment (sframba) [inline src/coreComponents/physicsSolvers/wavePropagation/AcousticFirstOrderWaveEquationSEM.hpp]: Please restore the old formatting of this comment

**Review by sframba — APPROVED** [review]

**Comment by castelletto1** [comment]
It seems that updated baselines are needed. Please confirm.

**Comment by acitrain** [comment]
> It seems that updated baselines are needed. Please confirm.

@castelletto1 Hum that's weird, I re-ran the integrated tests this morning on quartz and every tests passed and I didn't touch anything since that moment..

Update: I just re-ran the integrated tests on quartz and they all pass well + CI seems to give the same results so I think we seems good to go without a new baseline

**Comment by castelletto1** [comment]
@acitrain: if you look at the CI, you will see that some tests are failing (lines 4064-4068):

```
ATS SUMMARY February 15, 2024 15:51:33
FAILED:  6 python3(impermeableFault_smoke_01_2_restartcheck), python3(permeableFault_smoke_01_2_restartcheck), python3(elas3D_firstOrder_abc_smoke_01_2_restartcheck), python3(elas3D_firstOrder_fs_smoke_01_2_restartcheck), python3(elas3D_Q3_firstOrder_abc_smoke_01_2_restartcheck), python3(elas3D_Q3_firstOrder_fs_smoke_01_2_restartcheck)
TIMEOUT:  1 geosx(pennyShapedViscosityDominated_smoke_01_1_geos)
PASSED:   706
SKIPPED:  10
```

Specifically: 
- `elas3D_firstOrder_abc_smoke_01_2_restartcheck`
-  `elas3D_firstOrder_fs_smoke_01_2_restartcheck`
-  `elas3D_Q3_firstOrder_abc_smoke_01_2_restartcheck`
-  `elas3D_Q3_firstOrder_fs_smoke_01_2_restartcheck`

On quartz I do observe the same behavior.

**Comment by acitrain** [comment]
> @acitrain: if you look at the CI, you will see that some tests are failing (lines 4064-4068):
> 
> ```
> ATS SUMMARY February 15, 2024 15:51:33
> FAILED:  6 python3(impermeableFault_smoke_01_2_restartcheck), python3(permeableFault_smoke_01_2_restartcheck), python3(elas3D_firstOrder_abc_smoke_01_2_restartcheck), python3(elas3D_firstOrder_fs_smoke_01_2_restartcheck), python3(elas3D_Q3_firstOrder_abc_smoke_01_2_restartcheck), python3(elas3D_Q3_firstOrder_fs_smoke_01_2_restartcheck)
> TIMEOUT:  1 geosx(pennyShapedViscosityDominated_smoke_01_1_geos)
> PASSED:   706
> SKIPPED:  10
> ```
> 
> Specifically:
> 
> * `elas3D_firstOrder_abc_smoke_01_2_restartcheck`
> * `elas3D_firstOrder_fs_smoke_01_2_restartcheck`
> * `elas3D_Q3_firstOrder_abc_smoke_01_2_restartcheck`
> * `elas3D_Q3_firstOrder_fs_smoke_01_2_restartcheck`
> 
> On quartz I do observe the same behavior.

Hum okay I'll take a closer look, thanks

**Comment by castelletto1** [comment]
@acitrain I took care of the rebaselining. The `fail check`s were triggered because the original baselines didn't have `sigmaxzNp1AtReceivers`, `sigmayzNp1AtReceivers`, `sigmazzNp1AtReceivers`, `sigmaxyNp1AtReceivers`, `sigmaxxNp1AtReceivers`, `sigmayyNp1AtReceivers`.

**Comment by acitrain** [comment]
> @acitrain I took care of the rebaselining. The `fail check`s were triggered because the original baselines didn't have `sigmaxzNp1AtReceivers`, `sigmayzNp1AtReceivers`, `sigmazzNp1AtReceivers`, `sigmaxyNp1AtReceivers`, `sigmaxxNp1AtReceivers`, `sigmayyNp1AtReceivers`.

Hoo ok I see, sorry... Thank you @castelletto1 !

**Pull request merged**